// @botpress/cli@0.8.67/dist/code-generation/integration-schemas/configuration-module.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var configuration_module_exports = {};
__export(configuration_module_exports, {
  ConfigurationModule: () => ConfigurationModule
});
module.exports = __toCommonJS(configuration_module_exports);
var import_const = require("../const");
var import_generators = require("../generators");
var import_module = require("../module");
var strings = __toESM(require("../strings"));
class ConfigurationModule extends import_module.Module {
  static async create(configuration) {
    const { schema } = configuration;
    if (!schema) {
      return new ConfigurationModule({
        path: import_const.INDEX_FILE,
        exportName: "Configuration",
        content: "export type Configuration = Record<string, never>;"
      });
    }
    const name = "configuration";
    const exportName = strings.typeName(name);
    return new ConfigurationModule({
      path: import_const.INDEX_FILE,
      exportName,
      content: await (0, import_generators.jsonSchemaToTypeScriptType)(schema, exportName)
    });
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ConfigurationModule
});
