// @botpress/cli@0.8.67/dist/code-generation/integration-schemas/entities-module.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var entities_module_exports = {};
__export(entities_module_exports, {
  EntitiesModule: () => EntitiesModule,
  EntityModule: () => EntityModule
});
module.exports = __toCommonJS(entities_module_exports);
var import_bluebird = __toESM(require("bluebird"));
var import_generators = require("../generators");
var import_module = require("../module");
var strings = __toESM(require("../strings"));
class EntityModule extends import_module.Module {
  static async create(name, entity) {
    const entityName = name;
    const schema = entity.schema;
    const exportName = strings.typeName(entityName);
    const def = {
      path: `${name}.ts`,
      exportName,
      content: await (0, import_generators.jsonSchemaToTypeScriptType)(schema, exportName)
    };
    return new EntityModule(def);
  }
}
class EntitiesModule extends import_module.ReExportTypeModule {
  static async create(entities) {
    const entityModules = await import_bluebird.default.map(
      Object.entries(entities),
      async ([entityName, entity]) => EntityModule.create(entityName, entity)
    );
    const inst = new EntitiesModule({
      exportName: strings.typeName("entities")
    });
    inst.pushDep(...entityModules);
    return inst;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  EntitiesModule,
  EntityModule
});
